src/mbgl/style/conversion/function.cpp

```diff
@@ -70,6 +72,72 @@ std::unique_ptr<Expression> convertTokenStringToExpression(const std::string& so
     }
 }
 
+template <class T>
+optional<PropertyExpression<T>> convertFunctionToExpression(const Convertible& value, Error& error, bool convertTokens) {
+    auto expression = convertFunctionToExpression(expression::valueTypeToExpressionType<T>(), value, error, convertTokens);
+    if (!expression) {
+        return nullopt;
+    }
+
+    optional<T> defaultValue;
+
+    auto defaultValueValue = objectMember(value, "default");
+    if (defaultValueValue) {
+        defaultValue = convert<T>(*defaultValueValue, error);
+        if (!defaultValue) {
+            error.message = R"(wrong type for "default": )" + error.message;
+            return nullopt;
+        }
+    }
+
+    return PropertyExpression<T>(std::move(*expression), defaultValue);
+}
+
+template optional<PropertyExpression<AlignmentType>>
```

Comment:
Pretty tempting to use a macro here:

```C++
#define EXPLICIT_INSTANTIATION(T) template optional<PropertyExpression<T>> \
    convertFunctionToExpression<T>(const Convertible&, Error&, bool);

EXPLICIT_INSTANTIATION(AlignmentType);
...
```

Reply:
I refrained from adding this because it's tricky to get complex typenames (e.g. `std::array<float, 2>`) to work correctly in a macro context. You could probably swing something with `__VA_ARGS__`, but that's nonstandard.